Lesson25/Gokhan/filter.js

```diff
@@ -0,0 +1,422 @@
+];
+
+const sentencesWith10Words = sentencesExercise5.filter((sentences) => sentences.split(' ').length <= 10);
+console.log('Exercise 5 - ', sentencesWith10Words);
+
+
+// Exercise 6:
+// Given an array of products, filter out the products that are out of stock (where the quantity is 0).
+const productsExercise6 = [
+	{
+		name: 'Laptop',
+		brand: 'Apple',
+		price: 1999,
+		quantity: 0,
+	},
+	{
+		name: 'Smartphone',
+		brand: 'Samsung',
+		price: 899,
+		quantity: 20,
+	},
+	{
+		name: 'TV',
+		brand: 'LG',
+		price: 1499,
+		quantity: 15,
+	},
+	{
+		name: 'Headphones',
+		brand: 'Sony',
+		price: 299,
+		quantity: 30,
+	},
+	{
+		name: 'Smartwatch',
+		brand: 'Apple',
+		price: 399,
+		quantity: 25,
+	},
+	{
+		name: 'Camera',
+		brand: 'Canon',
+		price: 999,
+		quantity: 0,
+	},
+	{
+		name: 'Gaming Console',
+		brand: 'Microsoft',
+		price: 499,
+		quantity: 8,
+	},
+	{
+		name: 'Tablet',
+		brand: 'Samsung',
+		price: 649,
+		quantity: 18,
+	},
+	{
+		name: 'Smart Speaker',
+		brand: 'Amazon',
+		price: 99,
+		quantity: 0,
+	},
+	{
+		name: 'Fitness Tracker',
+		brand: 'Fitbit',
+		price: 129,
+		quantity: 14,
+	},
+];
+
+const emptyStock = productsExercise6.filter((stock) => stock.quantity <= 0);
+console.log('Exercise 6 - ', 'Empty Stocks Are: ', emptyStock);
+
+// Exercise 7:
+// Given an array of numbers, filter out the numbers that are divisible by 3.
+const numbersExercise7 = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10];
+
+const divisibleBy3 = numbersExercise7.filter((number) => number % 3 === 0);
+console.log('Exercise 7 - ', 'Divisible Numbers With 3: ', divisibleBy3);
+
+// Exercise 8:
+// Given an array of strings, filter out the strings that have more than 5 characters.
+const wordsExercise8 = [
+	'apple',
+	'banana',
+	'car',
+	'dog',
+	'elephant',
+	'flower',
+	'guitar',
+	'house',
+	'island',
+	'jungle',
+];
+
+const moreThan5Words = wordsExercise8.filter((words) => words.length <= 5);
+console.log('Exercise 8 - ', 'More Than 5 Characters Of Words: ', moreThan5Words);
+
+// Exercise 9:
+// Given an array of objects representing students, filter out the students who have a grade lower than C.
+const studentsExercise9 = [
+	{
+		id: 1,
+		name: 'Alice',
+		age: 18,
+		grade: 'A',
+	},
+	{
+		id: 2,
+		name: 'Bob',
+		age: 19,
+		grade: 'B',
+	},
+	{
+		id: 3,
+		name: 'Charlie',
+		age: 20,
+		grade: 'B+',
+	},
+	{
+		id: 4,
+		name: 'David',
+		age: 19,
+		grade: 'A-',
+	},
+	{
+		id: 5,
+		name: 'Emma',
+		age: 18,
+		grade: 'A+',
+	},
+	{
+		id: 6,
+		name: 'Frank',
+		age: 20,
+		grade: 'B',
+	},
+	{
+		id: 7,
+		name: 'Grace',
+		age: 19,
+		grade: 'D+',
+	},
+	{
+		id: 8,
+		name: 'Hannah',
+		age: 20,
+		grade: 'A-',
+	},
+	{
+		id: 9,
+		name: 'Isaac',
+		age: 18,
+		grade: 'D',
+	},
+	{
+		id: 10,
+		name: 'Julia',
+		age: 19,
+		grade: 'B-',
+	},
+];
+
+const gradeLowerThanC = studentsExercise9.filter((students) => students.grade >= 'C');
```

Comment:
You cannot compare letters reliably like this. There are different approaches to this, but one could be like this:
```suggestion
const passingGrades = ['A+', 'A', 'A-', 'B+', 'B', 'B-', 'C+', 'C'];
const gradeLowerThanC = studentsExercise9.filter((students) => !passingGrades.includes(students.grade.toUpperCase()));
```